src/main/scala/datalog/execution/ir/IROp.scala

```diff
@@ -14,8 +14,9 @@ import scala.quoted.*
 import scala.util.{Failure, Success}
 
 enum OpCode:
-  case PROGRAM, SWAP_CLEAR, SEQ, SCAN, SCANEDB, SPJ, INSERT, UNION, DIFF, DEBUG, DEBUGP, DOWHILE,
-  EVAL_RULE_NAIVE, EVAL_RULE_SN, EVAL_RULE_BODY, EVAL_NAIVE, EVAL_SN, LOOP_BODY, OTHER // convenience labels for generating functions
+  case PROGRAM, SWAP_CLEAR, SEQ, SCAN, SCANEDB, SPJ, INSERT, UNION, DIFF, 
+  DEBUG, DEBUGP, DOWHILE, UPDATE_DISCOVERED,
+  EVAL_STRATA, EVAL_STRATUM, EVAL_RULE_NAIVE, EVAL_RULE_SN, EVAL_RULE_BODY, EVAL_NAIVE, EVAL_SN, LOOP_BODY, OTHER // convenience labels for generating functions
```

Comment:
Why do you ned both EVAL STRATA and EVAL_STRATUM? Could you use only one extra label?